cpp/src/arrow/flight/flight_sql/CMakeLists.txt

```diff
@@ -87,7 +83,7 @@ if(ARROW_BUILD_TESTS OR ARROW_BUILD_EXAMPLES)
 
   add_executable(flight_sql_test_server
                  server.cc
-                 sql_info_util.cc
+                 sql_info_internal.cc
```

Comment:
Hang on. Why are we still linking the library source files into an executable?

Reply:
Even if the files are marked internal, you should still be able to get at the symbols during linking (you may need to ARROW_EXPORT the classes/functions/etc.)

Reply:
Ops... didn't need link it, fixed